graylog2-server/src/main/java/org/graylog2/contentpacks/facades/dashboardV1/DashboardWidgetConverter.java

```diff
@@ -0,0 +1,462 @@
+import org.graylog2.contentpacks.model.entities.WidgetConfig;
+import org.graylog2.contentpacks.model.entities.WidgetEntity;
+import org.graylog2.contentpacks.model.entities.references.ValueReference;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class DashboardWidgetConverter {
+    private Map<String, ValueReference> parameters;
+    private DashboardWidgetEntity dashboardWidgetEntity;
+    private WidgetConfig config;
+
+    public DashboardWidgetConverter(DashboardWidgetEntity dashboardWidgetEntity,
+                                    Map<String, ValueReference> parameters) {
+        this.dashboardWidgetEntity = dashboardWidgetEntity;
+        this.parameters = parameters;
+        this.config = new WidgetConfig(dashboardWidgetEntity.configuration(), parameters);
+    }
+
+    public static List<WidgetEntity> convert(DashboardWidgetEntity dashboardWidgetEntity,
+                                             Map<String, ValueReference> parameters) {
+        final DashboardWidgetConverter converter = new DashboardWidgetConverter(dashboardWidgetEntity, parameters);
+        return converter.convert();
+    }
+
+    public List<WidgetEntity> convert() {
+        final String type = dashboardWidgetEntity.type().asString(parameters);
+
+        try {
+            switch (type.toUpperCase(Locale.ENGLISH)) {
+                case "SEARCH_RESULT_CHART":
+                    return createHistogramWidget();
+                case "FIELD_CHART":
+                    return createFieldChartWidget();
+                case "STACKED_CHART":
+                    return createStackedChartWidget();
+                case "STATS_COUNT":
+                    return createStatsCountWidget();
+                case "QUICKVALUES":
+                    return createQuickValueWidgets();
+                case "STREAM_SEARCH_RESULT_COUNT":
+                case "SEARCH_RESULT_COUNT":
+                    return createSearchResultCount();
+                case "QUICKVALUES_HISTOGRAM":
+                    return createQuickValueHistogramWidgets();
+                case "ORG.GRAYLOG.PLUGINS.MAP.WIDGET.STRATEGY.MAPWIDGETSTRATEGY":
+                    return createMapWidget();
+                default: {
+                    throw new IllegalArgumentException(
+                            "The provided entity does not have a valid Widget type: " + type);
+                }
+            }
+        } catch (
+                InvalidRangeParametersException e) {
+            throw new IllegalArgumentException(
+                    "The provided entity does not have a valid timerange type: " + e.getMessage());
+        }
+    }
+
```

Comment:
I have read through the remainder of this class, but feel unable to really review it in depth in a reasonable time frame, as it's a huge bunch of special case logic for the various widget types. Gotta trust you here. :)